### PR TITLE
Add error message when a permission is missing

### PR DESCRIPTION
**Pull request by RainbowDashLabs**
Adds a error message in the channel when a permission is missing.

Sends a pn to the server owner, if the bot cant write in the channel.

### CODE REVIEW

File: src/main/java/de/chojo/repbot/ReputationBot.java [hunk 1]
```diff
@@ -193,8 +200,31 @@ private void initBot() {
...
+                            context.channel().sendMessage(errorMessage).queue();
+                            return;
+                        }
+                        var ownerId = context.guild().getOwnerIdLong();
+                        var finalErrorMessage = errorMessage;
+                        context.guild().retrieveMemberById(ownerId)
+                                .queue(member -> member.getUser().openPrivateChannel()
+                                        .queue(privateChannel -> privateChannel.sendMessage(finalErrorMessage)
+                                                .queue(succ -> {}, err -> ErrorResponseException.ignore(ErrorResponse.CANNOT_SEND_TO_USER))));
+                        return;
```
Comment (DRSchlaubi) [inline src/main/java/de/chojo/repbot/ReputationBot.java:225]: that's what flatMap and onErrorFlatMap is for

**Review by DRSchlaubi — CHANGES_REQUESTED** [review]

File: src/main/java/de/chojo/repbot/ReputationBot.java (hunk 1 above)
Comment (RainbowDashLabs) [reply src/main/java/de/chojo/repbot/ReputationBot.java:225]:
```java
context.guild().retrieveMemberById(ownerId)
    .flatMap(member -> member.getUser().openPrivateChannel())
    .flatMap(privateChannel -> privateChannel.sendMessage(finalErrorMessage))
    .onErrorMap(t -> null)
    .queue();
```
Something like this?

Comment (DRSchlaubi) [reply src/main/java/de/chojo/repbot/ReputationBot.java:225]: can't you do the ErrorResponse.ignore thing in the flat map

Comment (RainbowDashLabs) [reply src/main/java/de/chojo/repbot/ReputationBot.java:225]: No the onErrorMap needs to return a message. so I can only return null.

Comment (DRSchlaubi) [reply src/main/java/de/chojo/repbot/ReputationBot.java:225]: Fucking hell that's stupid ahhhhhhh.
No afaik it doesn't so either we return null 
Or we use the ugly queue we had before and be more explicit about the ignore

Comment (RainbowDashLabs) [reply src/main/java/de/chojo/repbot/ReputationBot.java:225]: the null return still looks better imo. I will keep this and push it.

**Review by dlsf — APPROVED** [review]

**Pull request merged**